homeassistant/components/samsungtv/media_player.py

```diff
@@ -110,14 +87,34 @@ def __init__(
         # Generate a configuration for the Samsung library
         self._config = {
             "name": "HomeAssistant",
-            "description": name,
+            "description": self._name,
             "id": "ha.component.samsung",
-            "method": method,
-            "port": port,
-            "host": host,
+            "method": config_entry.data[CONF_METHOD],
+            "port": config_entry.data.get(CONF_PORT),
+            "host": config_entry.data[CONF_HOST],
             "timeout": 1,
         }
 
+        # set up config_entry updates listener
+        config_entry.add_update_listener(self.async_config_entry_update)
```

Comment:
Register the listener in `async_added_to_hass`. Save the return value as an instance attribute and call it to unregister in `async_will_remove_from_hass`.

Reply:
Register `update_device_info` as the callback instead.